src/snowflake/cli/plugins/init/commands.py

```diff
@@ -0,0 +1,241 @@
+# limitations under the License.
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional
+
+import typer
+import yaml
+from click import ClickException
+from snowflake.cli.api.commands.flags import (
+    NoInteractiveOption,
+    VariablesOption,
+    parse_key_value_variables,
+)
+from snowflake.cli.api.commands.snow_typer import SnowTyperFactory
+from snowflake.cli.api.constants import DEFAULT_SIZE_LIMIT_MB
+from snowflake.cli.api.output.types import (
+    CommandResult,
+    MessageResult,
+)
+from snowflake.cli.api.project.schemas.template import Template, TemplateVariable
+from snowflake.cli.api.rendering.project_templates import render_template_files
+from snowflake.cli.api.secure_path import SecurePath
+
+# simple Typer with defaults because it won't become a command group as it contains only one command
+app = SnowTyperFactory()
+
+
+DEFAULT_SOURCE = "https://github.com/snowflakedb/snowflake-cli-templates"
+
+log = logging.getLogger(__name__)
+
+
+def _path_argument_callback(path: str) -> str:
+    if SecurePath(path).exists():
+        raise ClickException(
+            f"The directory {path} already exists. Please specify a different path for the project."
+        )
+    return path
+
+
+PathArgument = typer.Argument(
+    ...,
+    help="Directory to be initialized with the project. This directory must not already exist",
+    show_default=False,
+    callback=_path_argument_callback,
+)
+TemplateOption = typer.Option(
+    None,
+    "--template",
+    help="which template (subdirectory of --template-source) should be used. If not provided,"
+    " whole source will be used as the template.",
+    show_default=False,
+)
+SourceOption = typer.Option(
+    default=DEFAULT_SOURCE,
+    help=f"local path to template directory or URL to git repository with templates.",
+)
+
+TEMPLATE_METADATA_FILE_NAME = "template.yml"
+
+
+def _fetch_local_template(
+    template_source: SecurePath, path: Optional[str], destination: SecurePath
+) -> SecurePath:
+    """Copies local template to [dest] and returns path to the template root.
+    Ends with an error of the template does not exist."""
+
+    template_source.assert_exists()
+    template_origin = template_source / path if path else template_source
+    log.info("Copying local template from %s", template_origin.path)
+    if not template_origin.exists():
+        raise ClickException(
+            f"Template '{path}' cannot be found under {template_source}"
+        )
+
+    template_origin.copy(destination.path)
+    return destination / template_origin.name
+
+
+def _fetch_remote_template(
+    url: str, path: Optional[str], destination: SecurePath
+) -> SecurePath:
+    """Downloads remote repository template to [dest],
+    and returns path to the template root.
+    Ends with an error of the template does not exist."""
+    from git import rmtree as git_rmtree
+
+    # TODO: during nativeapp refactor get rid of this dependency
```

Comment:
Is this native app refactor planned already? My understanding of the plan is that we will keep "snow app init" for a while until:

1. we confirm that "snow init" supports all of our use cases
2. definition_version: 2 is out

Then we'll re-create all of the templates in the new repo with definition_version: 2. Does this align with the plan?

Reply:
Moved discussion to Slack